axelor-quality/src/main/java/com/axelor/apps/quality/service/QIAnalysisServiceImpl.java

```diff
@@ -35,4 +97,168 @@ public int setAdvancement(QIAnalysis qiAnalysis) {
     return totalAdvancement / qiTasksList.size();
   }
+
+  @Override
+  public List<QIActionDistribution> generateQIActionDistribution(QIAnalysis qiAnalysis)
+      throws AxelorException, IOException {
+
+    List<QITask> qiTasksList =
+        qiAnalysis.getQiTasksList().stream()
+            .filter(line -> line.isSelected())
+            .collect(Collectors.toList());
+
+    QualityConfig qualityConfig = qualityConfigRepository.all().fetchOne();
+    Sequence qiActionDistributionSequence =
+        (qualityConfig != null) ? qualityConfig.getQiActionDistributionSequence() : null;
+    if (qiActionDistributionSequence == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(QualityExceptionMessage.QI_ACTION_DISTRIBUTION_SEQUENCE_NOT_SET));
+    }
+
+    Map<Partner, List<QITask>> qiTasksGroupedByPartner =
+        qiTasksList.stream().collect(Collectors.groupingBy(QITask::getResponsiblePartner));
+    qiAnalysis = qiAnalysisRepository.find(qiAnalysis.getId());
+
+    List<QIActionDistribution> generatedQiActionDistributionList = new ArrayList<>();
+    for (Map.Entry<Partner, List<QITask>> entry : qiTasksGroupedByPartner.entrySet()) {
+      Partner responsiblePartner = entry.getKey();
+      List<QITask> qiTasks = entry.getValue();
+      createQIActionDistribution(
+          qiAnalysis,
+          qualityConfig,
+          qiActionDistributionSequence,
+          generatedQiActionDistributionList,
+          responsiblePartner,
+          qiTasks);
+    }
+    return generatedQiActionDistributionList;
+  }
+
+  @Override
+  public List<QIActionDistribution> generateQIActionDistributionForOthers(
+      QIAnalysis qiAnalysis, Integer recepient, Partner recepientPartner)
+      throws AxelorException, IOException {
+    qiAnalysis = qiAnalysisRepository.find(qiAnalysis.getId());
+    QualityConfig qualityConfig = qualityConfigRepository.all().fetchOne();
+    Sequence qiActionDistributionSequence =
+        (qualityConfig != null) ? qualityConfig.getQiActionDistributionSequence() : null;
+    if (qiActionDistributionSequence == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_CONFIGURATION_ERROR,
+          I18n.get(QualityExceptionMessage.QI_ACTION_DISTRIBUTION_SEQUENCE_NOT_SET));
+    }
+    List<QIActionDistribution> generatedQiActionDistributionListForOthers = new ArrayList<>();
+    createQIActionDistributionForOthers(
+        qiAnalysis,
+        recepient,
+        recepientPartner,
+        qualityConfig,
+        qiActionDistributionSequence,
+        generatedQiActionDistributionListForOthers);
+    return generatedQiActionDistributionListForOthers;
+  }
+
+  @Transactional(rollbackOn = Exception.class)
+  protected void createQIActionDistribution(
+      QIAnalysis qiAnalysis,
+      QualityConfig qualityConfig,
+      Sequence qiActionDistributionSequence,
```

Comment:
With the new qualifConfigService you can remove this parameter, and directly call the config directly from the injected service when we need them